pkg/cloudfabric-controller/controllerframework/controller_framework_test.go

```diff
@@ -290,7 +292,460 @@ func TestConsolidateControllerInstances_ReturnValues_MergeAndAutoExtends(t *test
 	assert.Equal(t, int64(math.MaxInt64), controllerInstanceBase.sortedControllerInstancesLocal[0].controllerKey)
 }
 
-func TestGetMaxInterval(t *testing.T) {
+// This test is to check while keep adding new controller manager instance, the scope will be evenly distributed
+func TestGenerateKeyContinuously(t *testing.T) {
+	controllerBase := new(ControllerBase)
+	controllerBase.sortedControllerInstancesLocal = []controllerInstanceLocal{}
+
+	// Totally add 1000 controller manager instance one by one
+	const TotalInstanceCount = 1000
+	for i := 0; i < TotalInstanceCount; i++ {
+		controllerKey, _ := controllerBase.generateKey()
+		controllerInstanceLocal := new(controllerInstanceLocal)
+		controllerInstanceLocal.controllerKey = controllerKey
+		generateTestSortedControllerInstances(controllerBase, controllerInstanceLocal)
+		verifyControllerKeyEvenlyDistributed(t, controllerBase)
+	}
+}
+
+func generateTestSortedControllerInstances(controllerBase *ControllerBase, controllerInstanceLocal *controllerInstanceLocal) {
+	// Generate controllerInstanceMap
+	controllerInstanceMap := make(map[string]v1.ControllerInstance)
+	index := 0
+	for _, instance := range controllerBase.sortedControllerInstancesLocal {
+		mapInstance := v1.ControllerInstance{
+			ControllerKey: instance.controllerKey,
+		}
+		controllerInstanceMap[string(index)] = mapInstance
+		index++
+	}
+	controllerInstanceMap[string(index)] = v1.ControllerInstance{
+		ControllerKey: controllerInstanceLocal.controllerKey,
+	}
+
+	sortedControllerInstancesLocal := SortControllerInstancesByKeyAndConvertToLocal(controllerInstanceMap)
+	_, _, _, _, _ = controllerBase.tryConsolidateControllerInstancesLocal(sortedControllerInstancesLocal)
+	controllerBase.sortedControllerInstancesLocal = sortedControllerInstancesLocal
+}
+
+func verifyControllerKeyEvenlyDistributed(t *testing.T, controllerBase *ControllerBase) {
+	count := len(controllerBase.sortedControllerInstancesLocal)
+	var totalSize int64
+	if count > 1 {
+		// In the total scope of 0 to math.MaxInt64, since 0 is count in size, total size is math.MaxInt64 + 1.
+		// To avoid int64 overflow, deduct one here for later assert.Equal.
+		totalSize--
+	}
+	sizeMap := make(map[int64]int)
+	for i := 0; i < count; i++ {
+		instance := controllerBase.sortedControllerInstancesLocal[i]
+		size := instance.Size()
+
+		if _, containsKey := sizeMap[size]; containsKey {
+			sizeMap[size]++
+		} else {
+			sizeMap[size] = 1
+		}
+		totalSize += size
```

Comment:
It seems unnecessary to sum totalSize as long as last controller has key MaxInt64

Reply:
I want to make sure all size is expected. For example, there is no overlap or broken scope between instances.